File: nprs_app/android/RecognitionSystem/src/main/java/com/nprs/recsystem/recognition/recognition_results/NumberPlate.java
package com.nprs.recsystem.recognition.recognition_results;

import com.nprs.recsystem.common.Rectangle;

public class NumberPlate {
    private final Rectangle _bounds;
    private final NumberPlateCharacter[] _characters;

    public NumberPlate(Rectangle bounds, NumberPlateCharacter[] characters) {
        _bounds = bounds;
        _characters = characters;
    }

    public Rectangle bounds() {
        return _bounds;
    }

    public NumberPlateCharacter[] characters() {
        return _characters;
    }
}

File: nprs_lib/sources/recognition_system/src/rec_system/common_structures/NumberPlate.cpp
#include "NumberPlate.h"

#include <iostream>

using namespace nprs;

NumberPlate::NumberPlate(
    const std::vector<sp<NumberPlateCharacter>> &characters, 
    const Quad &bounds,
    const Line &line,
    Rectangle rect)
    : _bounds(bounds), 
      _characters(characters),
      _line(line),
      _boundingRect(rect)
{
}

NumberPlate::NumberPlate(NumberPlate && other)
    : _bounds(other._bounds), 
      _characters(std::move(other._characters)),
      _line(other._line)
{
}

NumberPlate& NumberPlate::operator=(NumberPlate && other) {
    if (this != &other) {
        _bounds = other._bounds;
        _characters = std::move(other.characters());
        _line = other._line;
    }
    return *this;
}
